pkg/cmd/cli/cmd/version.go

```diff
@@ -0,0 +1,183 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	etcdversion "github.com/coreos/etcd/version"
+
+	"k8s.io/kubernetes/pkg/client/restclient"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	clientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	kubeversion "k8s.io/kubernetes/pkg/version"
+
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/cmd/util/tokencmd"
+	"github.com/openshift/origin/pkg/version"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	versionLong = `
+Display client and server versions.`
+)
+
+type VersionOptions struct {
+	BaseName         string
+	Config           clientcmdapi.Config
+	ClientConfig     *restclient.Config
+	OClient          *client.Client
+	KClient          *kclient.Client
+	Out              io.Writer
+	ClientCmdFactory *clientcmd.Factory
+
+	PrintEtcdVersion    bool
+	PrintClientFeatures bool
+}
+
+// NewCmdVersion creates a command for displaying the version of this binary
+func NewCmdVersion(fullName string, f *clientcmd.Factory, out io.Writer, options VersionOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "version",
+		Short: "Display client and server versions",
+		Long:  versionLong,
+		Run: func(cmd *cobra.Command, args []string) {
+			options.BaseName = fullName
+
+			if err := options.Complete(f, args, out); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(cmd, err.Error()))
+			}
+
+			if err := options.RunVersion(); err != nil {
+				kcmdutil.CheckErr(err)
+			}
+		},
+	}
+
+	return cmd
+}
+
+func (o *VersionOptions) Complete(f *clientcmd.Factory, args []string, out io.Writer) error {
+	if len(args) > 0 {
+		return fmt.Errorf("no arguments should be passed")
+	}
+
+	o.Out = out
+	o.ClientCmdFactory = f
+
+	return nil
+}
+
+// RunVersion attempts to display client and server versions for Kubernetes and OpenShift
+func (o VersionOptions) RunVersion() error {
+	if o.ClientCmdFactory != nil {
+		fmt.Fprintf(o.Out, "%s\n", "CLIENTS")
+	}
+
+	fmt.Fprintf(o.Out, "%s %v\n", o.BaseName, version.Get())
+	fmt.Fprintf(o.Out, "kubernetes %v\n", kubeversion.Get())
+	if o.PrintEtcdVersion {
+		fmt.Fprintf(o.Out, "etcd %v\n", etcdversion.Version)
```

Comment:
@deads2k 
> Why would I want to print the local etcd version in the client?
> Seems like you want another server version we don't yet expose here. Do this later?

Okay, I just kept it from [pkg/version/version.go](https://github.com/openshift/origin/blob/master/pkg/version/version.go#L107), since `pkg/cmd/openshift/openshift.go` made a call to this function, requesting `etcd version` here: https://github.com/openshift/origin/blob/master/pkg/cmd/openshift/openshift.go#L119